docs/sources/operators-guide/configure/configure-helm-ha-deduplication-consul/index.md

```diff
@@ -0,0 +1,133 @@
+
+Grafana Mimir can deduplicate data from a high-availability (HA) Prometheus setup. You can configure
+the deduplication by using the Grafana Mimir helm chart deployment that uses external Consul. For more information, see [Configure high availability]({{< relref "../configure-high-availability-deduplication.md" >}}).
+
+## Before you begin
+
+You need to have a Grafana Mimir installed.
+
+For conceptual information about how Mimir deduplicates incoming HA samples, refer to [Configure high availability]({{< relref "../configure-high-availability-deduplication.md" >}}).
+
+You also have to configure HA for Prometheus or Grafana Agent. Last, you need a KV store. In this guide you will
+use Consul. You will be guided on the setup if you haven't had one.
+
+## Configure Prometheus or Grafana Agent to send HA external labels
+
+Configure Prometheus or Grafana Agent HA setup by setting label called `cluster` and `__replica__`. These two labels
+name are default labels for HA setup in Grafana Mimir. If you want to change the labels, make sure to change it in Mimir
+too to make Grafana Mimir and Prometheus/Grafana Agent config matches with each other, otherwise HA deduplication
+wouldn't work. `cluster` label value must be same across replica belong to the same cluster. `__replica__` must be
+unique across different replica in the cluster.
+
+```yaml
+global:
+  scrape_interval: 1m
+  scrape_timeout: 10s
+  evaluation_interval: 1m
+  external_labels:
+    __replica__: replica-1
+    cluster: my-prometheus
+```
+
+Reload or restart Prometheus or Grafana Agent after updating the configuration.
+
+## Install Consul using Helm
+
+To get and install Consul on Kubernetes, use the [Consul helm chart](https://github.com/hashicorp/consul-k8s/tree/main/charts/consul).
+Make a note of the Consul endpoint, because you will need it later to configure Mimir.
+
+## Configure Mimir high availability
+
+You can configure Mimir HA deduplication globally or at the tenant level.
+
+### Globally configure HA deduplication
+
+Before you begin, make sure that Mimir was installed using the Helm chart.
+
+1. Add the following configuration to your `custom.yaml` file:
+
+```yaml
+mimir:
+  structuredConfig:
+    limits:
+      accept_ha_samples: true
+      # The following two configurations must match those of external_labels in Prometheus
+      # The config values below are the default and can be removed if you don't want to override to a new value
+      ha_cluster_label: cluster
+      ha_replica_label: __replica__
+    distributor:
+      ha_tracker:
+        enable_ha_tracker: true
+        kvstore:
+          store: consul
+          consul:
+            host: <consul-endpoint> # example: http://consul.consul.svc.cluster.local:8500
+```
+
+2. Upgrade the Mimir's helm release using the following command:
+
+```bash
+ helm -n <mimir-namespace> upgrade --install mimir grafana/mimir-distributed -f custom.yaml
+```
+
+### Configure HA deduplication per tenant
+
```

Comment:
I think we should mention that you need to set up the distributor in this case as well.
Maybe add to the yaml the structuredConfig part about the distriburtor.

Reply:
Updated with adding the complete structuredConfig